osbenchmark/builder/downloaders/builders/source_builder.py

```diff
@@ -0,0 +1,40 @@
+import logging
+import os
+
+from osbenchmark.builder.utils.jdk_resolver import JdkResolver
+from osbenchmark.builder.utils.path_manager import PathManager
+from osbenchmark.exceptions import ExecutorError, BuildError
+
+
+class SourceBuilder:
+    def __init__(self, executor, source_directory, build_jdk, log_directory):
+        self.logger = logging.getLogger(__name__)
+        self.executor = executor
+        self.path_manager = PathManager(executor)
+        self.jdk_resolver = JdkResolver(executor)
```

Comment:
General comment : How about we use factory pattern here to avoid hard bounding concrete dependencies?

Reply:
That's a good idea, will refactor